mailet/standard/src/main/java/org/apache/james/transport/matchers/AtMost.java

```diff
@@ -33,39 +35,71 @@
 import org.apache.mailet.base.GenericMatcher;
 import org.apache.mailet.base.MailetUtil;
 
+import com.google.common.base.Preconditions;
+import com.google.common.base.Splitter;
 import com.google.common.collect.ImmutableList;
 
 /**
- * Checks that a mail did at most X tries on a specific operation.
+ * <p>Checks that a mail did at most X retries on a specific operation</p>
  *
- * If no retries have been performed previously, it sets up an attribute `AT_MOST_TRIES`
- * in the mail that will be incremented everything the check succeeds.
+ * <p> If no retries have been performed previously for Y attribute, it'll be sets up.</p>
+ * <p> In the mail, every time the check was succeeds, counter of it will be increment by one.
+ * The check fails when the defined X limit is reached.</p>
  *
- * The check fails when the defined X limit is reached.
+ * <ul>
+ * <li>X - count of how many times a specific operation retried</li>
+ * <li>Y - name of attribute represented for specific operation retried, default value is: <i>AT_MOST_TRIES</i></li>
+ * </ul>
  *
- * <p>The example below will match mail with at most 3 tries on the mailet</p>
+ * <p>The example below will match mail with at most 3 tries on the mailet
+ * with attribute name <i>AT_MOST_TRIES</i></p>
  *
  * <pre><code>
- * &lt;mailet match=&quot;AtMost=3&quot; class=&quot;&lt;any-class&gt;&quot;&gt;
+ * &lt;mailet match=&quot;AtMost=AT_MOST_TRIES:3&quot; class=&quot;&lt;any-class&gt;&quot;&gt;
```

Comment:
Same comment as on #3546. But you speak of number of retries and here we configure a number of tries. Not clear

Reply:
Yes, this PR is a next step for it.